modules/salt/pages/salt-rate-limiting.adoc

```diff
@@ -0,0 +1,61 @@
+[[salt.rate.limiting]]
+= Rate Limiting
+
+Salt is able to run commands in parallel on a large number of minions.
+This can potentially create large amounts of load on your infrastructure.
+You can use these rate-limiting parameters to control the load in your environment.
+
+These parameters are all configured in the [filename]``/etc/rhn/rhn.conf`` configuration file.
+
+[NOTE]
+====
+Salt commands that are executed from the command line are not subject to these parameters.
+====
+
+
+
+== Batching
+
+There are two parameters that control how actions are sent to clients, one for the batch size, and one for the delay.
+
+When the Salt master sends a batch of actions to the target minions, it will send it to the number of minions determined in the batch size parameter.
+As each minion completes, the action begins on a new minion after the specified delay period.
+
+Choosing a lower batch size will reduce system load and parallelism, but might reduce overall performance for processing actions.
+
+The batch size parameter sets the maximum number of clients that can execute a single action at the same time.
+Adjust the [systemitem]``java.salt_batch_size`` parameter.
+Defaults to 100.
+
+Increasing the delay increases the chance that multiple minions will have completed before the next action is issued, resulting in fewer overall commands, and reducing load.
+
+The batch delay parameter sets the amount of time, in seconds, to wait after an action is processed before beginning to process the action on the next minion.
+Adjust the [systemitem]``java.salt_batch_delay`` parameter.
+Defaults to 1.0 seconds.
+
+
+
+== Presence Ping Timeout
+
+There are two parameters that control how presence pings from the Salt master are handled, one for the ping timeout, and one for the ping gather job.
+
+Salt batch calls begin with the Salt master performing a presence ping on the target minions.
+A ping gather job runs on the Salt master to handle the incoming pings from the minions.
+Batched actions will begin only after all minions have either responded to the ping, or timed out.
+
+The presence ping is an ordinary Salt command, so it is subject to the same timeout parameters as all other Salt commands.
```

Comment:
Hmmm, not exactly. I don't know how to put this in understandable English, but in my tech-English it sounds like "it is subject to the same *kind* of timeout parameters, albeit with different values".

In concrete terms: `salt_presence_ping_timeout` and `salt_presence_ping_gather_job_timeout` override `timeout` and `gather_job_timeout` respectively, but only for presence pings.

More below.

Reply:
I had interpreted `timeout` and `gather_job_timeout` as global variables, and the  presence_ping  ones as specific variables that override the  global ones. Am I misunderstanding?

Reply:
That is correct. Let me give a try to rephrase the sentence above to try to show what I wanted to say:

```suggestion
The presence ping is an ordinary Salt command, but is not subject to the same timeout parameters as all other Salt commands (`timeout`/`gather_job_timeout`), rather, it has its own parameters (`presence_ping_timeout`/`presence_ping_gather_job_timeout`).
```